src/cargo/ops/cargo_rustc/compilation.rs

```diff
@@ -59,8 +61,8 @@ impl Compilation {
     ///
     /// The package argument is also used to configure environment variables as
     /// well as the working directory of the child process.
-    pub fn process<T: ToCStr>(&self, cmd: T, pkg: &Package)
-                              -> CargoResult<util::ProcessBuilder> {
+    pub fn process(&self, cmd: CommandType, pkg: &Package)
+                   -> CargoResult<CommandPrototype> {
```

Comment:
This is already not exactly super-ergonomic to call, could some helper methods for things like `rustc`, `rustdoc`, and `custom` be added to cut down on the boilerplate?